src/pudl/transform/vcerare.py

```diff
@@ -0,0 +1,399 @@
+
+    There are a handful of rows that are not counties but rather
+    cities or portions of lakes. We drop the city values in
+    :func:`_drop_city_records`, but we keep the lake values.
+    Because these are not within county geographies, we will
+    not assign them FIPS codes.
+    """
+    logger.info("Nulling FIPS IDs for lake rows")
+    lake_county_state_names = [
+        "lake_erie_ohio",
+        "lake_hurron_michigan",
+        "lake_michigan_illinois",
+        "lake_michigan_indiana",
+        "lake_michigan_michigan",
+        "lake_michigan_wisconsin",
+        "lake_ontario_new_york",
+        "lake_st_clair_michigan",
+        "lake_superior_minnesota",
+        "lake_superior_michigan",
+        "lake_superior_wisconsin",
+    ]
+    # Null county_id_fips codes
+    df.loc[df["county_state_names"].isin(lake_county_state_names), "county_id_fips"] = (
+        pd.NA
+    )
+    return df
+
+
+@asset(
+    io_manager_key="parquet_io_manager",
+    compute_kind="pandas",
+    op_tags={"memory-use": "high"},
+)
+def out_vcerare__hourly_available_capacity_factor(
+    raw_vcerare__lat_lon_fips: pd.DataFrame,
+    raw_vcerare__fixed_solar_pv_lat_upv: pd.DataFrame,
+    raw_vcerare__offshore_wind_power_140m: pd.DataFrame,
+    raw_vcerare__onshore_wind_power_100m: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform raw Vibrant Clean Energy renewable generation profiles.
+
+    Concatenates the solar and wind capacity factors into a single table and turns
+    the columns for each county or subregion into a single county_or_lake_name column.
+    """
+    logger.info("Transforming the hourly available capacity factor tables")
+    # Clean up the FIPS table
+    fips_df = _prep_lat_long_fips_df(raw_vcerare__lat_lon_fips)
+    # Apply the same transforms to all the capacity factor tables. This is slower
+    # than doing it to a concatenated table but less memory intensive because
+    # it doesn't need to process the ginormous table all at once.
+    raw_dict = {
+        "solar_pv": raw_vcerare__fixed_solar_pv_lat_upv,
+        "offshore_wind": raw_vcerare__offshore_wind_power_140m,
+        "onshore_wind": raw_vcerare__onshore_wind_power_100m,
+    }
+    clean_dict = {
+        df_name: _check_for_valid_counties(df, fips_df, df_name)
+        .pipe(_add_time_cols, df_name)
+        .pipe(_make_cap_fac_frac, df_name)
+        .pipe(_stack_cap_fac_df, df_name)
+        for df_name, df in raw_dict.items()
+    }
+    # Combine the data and perform a few last cleaning mechanisms
+    return (
+        _combine_all_cap_fac_dfs(clean_dict)
+        .pipe(_drop_city_records)
+        .pipe(_combine_cap_fac_with_fips_df, fips_df)
+        .pipe(_null_non_county_fips_rows)
+    )
+
+
+@asset_check(
+    asset=out_vcerare__hourly_available_capacity_factor,
+    blocking=True,
+    description="Check that output table is as expected.",
+)
+def check_hourly_available_cap_fac_table(asset_df: pd.DataFrame):  # noqa: C901
```

Comment:
Not a blocker, but with a lot of different checks inside a single function like this, we probably want to accumulate some kind of error report and then at the end of the function, if there are any errors in that report, we return the `AssetCheckResult` and include all of that information, rather than bailing as soon as we find any failure. Otherwise if there are multiple failures you'll have to run the whole asset check multiple times to find them all.